### PR TITLE
Removed legacy pipelines & refactored the resolution context.

### PR DESCRIPTION
**Pull request by iwoplaza**

### CODE REVIEW

File: packages/typegpu/src/core/texture/texture.ts [hunk 1]
```diff
@@ -537,7 +520,19 @@ class TgpuBindableSampledTextureImpl
...
     const id = ctx.names.makeUnique(this.label);
-    ctx.addRenderResource(this, id);
+    const { group, binding } = ctx.allocateFixedEntry(this);
+
+    let type: string;
+    if ((this._texture.props.sampleCount ?? 1) > 1) {
+      type = 'texture_multisampled_2d';
+    } else {
+      type = `texture_${dimensionToCodeMap[this.dimension]}`;
+    }
```
Comment (mhawryluk) [inline packages/typegpu/src/core/texture/texture.ts]: replace with const and ternany operator

File: packages/typegpu/src/tgpuBindGroupLayout.ts [hunk 2]
```diff
@@ -528,28 +569,10 @@ class TgpuBindGroupImpl<
...
-            }
-
-            return {
-              binding: idx,
-              resource,
-            };
-          }
-
-          if ('sampler' in entry) {
+          if ('externalTexture' in entry || 'sampler' in entry) {
```
Comment (mhawryluk) [inline packages/typegpu/src/tgpuBindGroupLayout.ts:572]: why don't we separate it to return GPUExternalTexture for externalTexture and GPUSampler for sampler?

File: packages/typegpu/src/resolutionCtx.ts [hunk 3]
```diff
@@ -366,50 +307,29 @@ export class ResolutionCtxImpl implements ResolutionCtx {
...
-
-    this.addDeclaration(
-      `@group(${group}) @binding(${idx}) var<${usageToVarTemplateMap[bindable.usage]}> ${identifier}: ${this.resolve((bindable.allocatable as TgpuBuffer<AnyTgpuData>).dataType)};`,
-    );
+    this._declarations.push(declaration);
   }
 
-  addRenderResource(resource: TgpuRenderResource, identifier: string): void {
-    const { group, idx } = this._shared.reserveRenderResourceEntry(resource);
+  allocateLayoutEntry(layout: TgpuBindGroupLayout): string {
```
Comment (mhawryluk) [inline packages/typegpu/src/resolutionCtx.ts:313]: maybe we could write some tests for this group index resolution process

**Review by mhawryluk — APPROVED** [review]
awesome! 🔪

File: packages/typegpu/src/tgpuBindGroupLayout.ts (hunk 2 above)
Comment (iwoplaza) [reply packages/typegpu/src/tgpuBindGroupLayout.ts:572]: Its the same JS behavior in both branches. We will probably split them later down the road when we refactor typed samplers.

**Pull request merged**